content/docs/user-guide/settings/developer-api.md

```diff
@@ -0,0 +1,401 @@
+
+The following example shows how to use the `SettingsRegistryInterface::GetObject` function to load the SettingsRegistry into an `AzPhysics::SceneConfiguration`.
+
+```c++
+// Make sure the AzPhysics::SceneConfiguration class is reflected
+void SceneConfiguration::Reflect(AZ::ReflectContext* context)
+{
+    Physics::WorldConfiguration::Reflect(context);
+
+    if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+    {
+        serializeContext->Class<SceneConfiguration>()
+            ->Version(1)
+            ->Field("LegacyConfig", &SceneConfiguration::m_legacyConfiguration)
+            ->Field("LegacyId", &SceneConfiguration::m_legacyId)
+            ->Field("Name", &SceneConfiguration::m_sceneName)
+            ;
+    }
+}
+
+// ... Later on the SceneConfiguration class can be loaded from the SettingsRegistry using the `GetObject` function
+AzPhysics::SceneConfiguration sceneConfig;
+bool configurationRead = false;
+
+AZ::SettingsRegistryInterface* settingsRegistry = AZ::SettingsRegistry::Get();
+if (settingsRegistry)
+{
+    configurationRead = settingsRegistry->GetObject(sceneConfig, "/Amazon/Gems/PhysX/DefaultSceneConfiguration");
+}
+```
+
+### Visitor API
+
+The [visitor API](https://github.com/o3de/o3de/blob/02846cf44347cbf4fae0faacc4a2ba74284908ff/Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h#L185-L194) supports recursively visiting the JSON object and array children starting at a specified JSON pointer. The visitor API allows for processing of each child field which isn't available when using the `GetObject` API. It also provides type flexibility when dealing with the JSON data. A visitor for the settings registry must implement the `AZ::SettingsRegistryInterface::Visitor` class. An instance of that class can be supplied to the `AZ::SettingsRegistryInterface::Visit()` method.
+
+The visitor is recommended when serializing in/out complex objects that don't have any `SerializeContext` reflection. Since it isn't tied to the `SerializeContext`, the same logic could be used to read objects from the Settings Registry using the JSON facilities available within other programming languages such as Python, JavaScript, or C#.
+
+The following example demonstrates using the visitor API to gather active gem info which is populated under the settings object at "/O3DE/Gems".  This is done by the [GetGemsInfo](https://github.com/o3de/o3de/blob/adb37ad54d69bcef23c5b7f70c77e669f8202194/Code/Framework/AzFramework/AzFramework/Gem/GemInfo.cpp#L27-L64) function.
+
+```c++
+// Queries the settings registry to get the list active gems targets and source paths
+auto GemSettingsVisitor = [&settingsRegistry, &gemInfoList]
+(const AZ::SettingsRegistryInterface::VisitArgs& gemVisitArgs)
```

Comment:
I'll combine the lines.
Visual Studio formats the code itself in [this manner](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Gem/GemInfo.cpp#L31-L32) and I was trying to keep it consistent.

But since we don't have to worry about long lines in documentation, combining the lines seems fine.